### DISCUSSION TITLE
Websocket can not Connect after new installation.

### DISCUSSION THREAD

**Discussion by Mozan00** [amidaware/tacticalrmm#1813] [General]
Hey guys, I've been trying to install TacticalRMM the traditional way for a few days now. I'm having some issues with figuring out how to use self-signed full-chain certificates because my firewall already has a reverse proxy that uses its own certificates. So far, I've completed the installation, and I can log in to rmm.domain.com and mesh.domain.com. However, when loading api.domain.com, I get "Not Found - The requested resource was not found on this server." Additionally, logging into Mesh Central says that the WebSocket cannot connect, and the agents won't install. Any help would be appreciated.

**Comment by dinger1986** [comment]
Hello, the API "Not Found" is meant to come up, to be clear did you install using insecure or another way? becaus you cant change from insecure to using a proxy after

**Reply by Mozan00** [reply]
I did not install using insecure.

**Reply by dinger1986** [reply]
Are you using your other proxy? Have you set it up as per our unsupported docs for proxys?

**Reply by Mozan00** [reply]
I installed Tacticall rmm the traditional way and it installed nginx. Nginx is working without errors.

**Reply by dinger1986** [reply]
so why do you mention your firewall and its certificates?

**Reply by Mozan00** [reply]
Well we have alot of Service behind our public ip and we use our firewall built proxy that comes with it is own certificates to direct pakets to the different services. Thats why i used self signed certificates. because in the end the firewall will always force it is own certificates. And that is okay. it is working with all other services. I dont know if the current Problem with trmm has something to do with nginx or not, because i seem to reach rmm mesh and api but the websocket is not connecting as i mentioned

**Reply by dinger1986** [reply]
so you are using another proxy? have you looked at the docs on setting it up? you will need either the cert installed or remove nginx on the trmm server and do all the proxying on your proxy but you cannot use self signed on the trmm server and hope that the other proxy applies certs correctly

**Reply by dinger1986** [reply]
here it is in the docs https://docs.tacticalrmm.com/unsupported_proxies/

**Reply by Mozan00** [reply]
So should i install it again without any certificates and without letting it install nginx and let the firewall proxy and subdomains just point at the server or can i just remove nginx now without a reinstall.

**Reply by dinger1986** [reply]
follow the docs and do it as it says

**Reply by Mozan00** [reply]
Just an update: I tested Mesh locally before setting up the second firewall reverse proxy, and it seems to be working. So the problem has to be with the firewall. Unfortunately, in my case, our firewall appears not to support WebSockets, which is why it's not working in the first place.

**Reply by dinger1986** [reply]
ok, so its your firewall at fault then, maybe worth looking at other proxies?

**Reply by Mozan00** [reply]
Well, what I can do is forward three random ports in the Firewall to the TRMM Server and configure nginx to listen on these ports and then forward them to port 443. But would that even work? Because the agents require port 443, don't they?

**Reply by dinger1986** [reply]
It can work on other ports but it's a lot of messing around.

Best to look at alternative proxys for your network as I'm sure other systems you will use will use websockets
